runtime/jvmti/suspendhelper.cpp

```diff
@@ -49,7 +49,7 @@ suspendThread(J9VMThread *currentThread, jthread thread, BOOLEAN allowNull, BOOL
 	 * suspended threads without invoking getVMThread -> acquireVThreadInspector. This approach
 	 * can be taken for both platform and virtual threads.
 	 */
-	if (0 != J9OBJECT_U32_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset)) {
+	if (J9_ARE_ANY_BITS_SET(J9OBJECT_U64_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset), 1)) {
```

Comment:
for better readability, nested calls should be avoided. this sequence is repeated multiple times, a helper should be added for it.
````
U_64 suspendedInternal = J9OBJECT_U64_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset);
if (J9_ARE_ANY_BITS_SET(suspendedInternal, 0x1)) {